yoyodyne/models/transducer.py

```diff
@@ -488,19 +488,16 @@ def log_sum_softmax_loss(
         normalization_term = torch.logsumexp(logits, -1)
         return log_sum_exp_terms - normalization_term
 
-    @staticmethod
-    def loss_func(
-        prediction: Tuple[List[List[int]], torch.Tensor],
-        target: torch.Tensor,
-    ) -> torch.Tensor:
-        """Returns loss value for prediction.
+    def get_loss_func(
+        self, reduction: str
+    ) -> Callable[[torch.Tensor, torch.Tensor], torch.Tensor]:
+        """Returns loss function for prediction.
 
         This model already produces prediction strings and calculates loss
-        during training. This simply serves as a wrapper to allow compatibility
-        with training_step in the superclass.
+        during training. This simply serves as a wrapper to prevent
+        superclass from creating an unused loss function.
         """
-        _, loss = prediction
-        return loss
+        return lambda predictions, target: predictions[1]
```

Comment:
Not sure how this is usually handled, but it strikes me as a bit odd that this is conceptually returns a loss function, but we have no docstring for it. If I am a developer making changes to `Transducer` is it clear how to interact with the loss function?

Reply:
This should be more coherrent in the other PR. Lemme know if you still see issues. Dev work should never touch the loss function since it's part o the forward pass.